chxtools/Circular_Average.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+
+
+def radial_grid(center, shape, pixel_size=None):
+    """Convert a cartesian grid (x,y) to the radius relative to some center
+
+    Parameters
+    ----------
+    center : tuple
+        point in image where r=0; may be a float giving subpixel precision.
+        Order is (rr, cc).
+    shape : tuple
+        Image shape which is used to determine the maximum extent of output
+        pixel coordinates.
+        Order is (rr, cc).
+    pixel_size : sequence, optional
+        The physical size of the pixels.
+        len(pixel_size) should be the same as len(shape)
+        defaults to (1,1)
+
+    Returns
+    -------
+    r : array
+        The distance of each pixel from `center`
+        Shape of the return value is equal to the `shape` input parameter
+    """
+
+    if pixel_size is None:
+        pixel_size = (1, 1)
+
+    X, Y = np.meshgrid(pixel_size[1] * (np.arange(shape[1]) - center[1]),
+                       pixel_size[0] * (np.arange(shape[0]) - center[0]))
+    return np.sqrt(X*X + Y*Y)
+
+
+def bin_1D(x, y, nx=None, min_x=None, max_x=None):
+    """
+    Bin the values in y based on their x-coordinates
+
+    Parameters
+    ----------
+    x : array
+        position
+    y : array
+        intensity
+    nx : integer, optional
+        number of bins to use defaults to default bin value
+    min_x : float, optional
+        Left edge of first bin defaults to minimum value of x
+    max_x : float, optional
+        Right edge of last bin defaults to maximum value of x
+
+    Returns
+    -------
+    edges : array
+        edges of bins, length nx + 1
+
+    val : array
+        sum of values in each bin, length nx
+
+    count : array
+        The number of counts in each bin, length nx
+    """
+
+    # handle default values
+    if min_x is None:
+        min_x = np.min(x)
+    if max_x is None:
+        max_x = np.max(x)
+    if nx is None:
+        nx = int(max_x - min_x)
+
+    # use a weighted histogram to get the bin sum
+    bins = np.linspace(start=min_x, stop=max_x, num=nx+1, endpoint=True)
+    val, _ = np.histogram(a=x, bins=bins, weights=y)
+    # use an un-weighted histogram to get the counts
+    count, _ = np.histogram(a=x, bins=bins)
+    # return the three arrays
+    return bins, val, count
+
+def bin_edges_to_centers(input_edges):
```

Comment:
@yugangzhang isn't this the same function in skbeam/utils bin_edges_to_centers function?